pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,446 @@
+
+package monitortool
+
+import (
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv4"
+	"golang.org/x/net/ipv6"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	config "antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha1"
+)
+
+var (
+	icmpSeq    uint32
+	icmpEchoID = os.Getpid() & 0xffff
+)
+
+const (
+	// For privileged
+	IPv4ProtocolICMPRaw = "ip4:icmp"
+	IPv6ProtocolICMPRaw = "ip6:ipv6-icmp"
+	IPProtocol          = "ip"
+	// ICMP protocol number.
+	ProtocolICMP   = 1
+	ProtocolICMPv6 = 58
+)
+
+// getICMPSeq returns the next sequence number as uint16,
+// wrapping around to 0 after reaching the maximum value of uint16.
+func getICMPSeq() uint16 {
+	// Increment the sequence number atomically and get the new value.
+	// We use atomic.AddUint32 and pass 1 as the increment.
+	// The returned value is the new value post-increment.
+	newVal := atomic.AddUint32(&icmpSeq, 1)
+
+	return uint16(newVal)
+}
+
+// NodeLatencyMonitor is a tool to monitor the latency of the Node.
+type NodeLatencyMonitor struct {
+	// latencyStore is the cache to store the latency of each Nodes.
+	latencyStore *LatencyStore
+	// latencyConfig is the config for the latency monitor.
+	latencyConfig *LatencyConfig
+	// latencyConfigChanged is the channel to notify the latency config changed.
+	latencyConfigChanged chan struct{}
+	// isIPv4Enabled is the flag to indicate whether the IPv4 is enabled.
+	isIPv4Enabled bool
+	// isIPv6Enabled is the flag to indicate whether the IPv6 is enabled.
+	isIPv6Enabled bool
+
+	nodeInformer               coreinformers.NodeInformer
+	nodeLatencyMonitorInformer crdinformers.NodeLatencyMonitorInformer
+}
+
+// LatencyConfig is the config for the latency monitor.
+type LatencyConfig struct {
+	// Enable is the flag to enable the latency monitor.
+	Enable bool
+	// Interval is the interval time to ping all Nodes.
+	Interval time.Duration
+}
+
+// NewNodeLatencyMonitor creates a new NodeLatencyMonitor.
+func NewNodeLatencyMonitor(nodeInformer coreinformers.NodeInformer,
+	nlmInformer crdinformers.NodeLatencyMonitorInformer,
+	nodeConfig *config.NodeConfig,
+	trafficEncapMode config.TrafficEncapModeType) *NodeLatencyMonitor {
+	m := &NodeLatencyMonitor{
+		latencyStore:               NewLatencyStore(trafficEncapMode.IsNetworkPolicyOnly()),
+		latencyConfig:              &LatencyConfig{Enable: false},
+		latencyConfigChanged:       make(chan struct{}, 1),
+		nodeInformer:               nodeInformer,
+		nodeLatencyMonitorInformer: nlmInformer,
+	}
+
+	isIPv4Enabled, _ := config.IsIPv4Enabled(nodeConfig, trafficEncapMode)
+	isIPv6Enabled, _ := config.IsIPv6Enabled(nodeConfig, trafficEncapMode)
+	m.isIPv4Enabled = isIPv4Enabled
+	m.isIPv6Enabled = isIPv6Enabled
```

Comment:
I think that there was no harm in logging the error here, even if @Dyanngg pointed out that it should not be possible. That being said, I don't feel strongly about it.
If you keep the current version, maybe make it less verbose as follows:
```go
m.isIPv4Enabled, _ = config.IsIPv4Enabled(nodeConfig, trafficEncapMode)
m.isIPv6Enabled, _ = config.IsIPv6Enabled(nodeConfig, trafficEncapMode)
```

Reply:
Thanks, I will change it.